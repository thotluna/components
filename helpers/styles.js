import classNames from 'classnames'

export const getDynamicClasses = (cssModule = {}, props = {}, classes) => {
  return classes.reduce((classesObject, classKey) => {
    const propValue = props[classKey]
    const className = cssModule[`${classKey}-${propValue}`]

    return className && propValue
      ? { ...classesObject, [className]: propValue }
      : classesObject
  }, {})
}

export const getModuleClasses = (cssModule, classKey) => {
  return (cssModule && cssModule[classKey]) || classKey
}

export const getObjectClasses = (cssModule, object) => {
  return Object.keys(object).reduce((classes, classKey) => {
    const className = cssModule[classKey]
    return className ? { ...classes, [className]: object[classKey] } : classes
  }, {})
}

export const getClasses = (cssModule = {}) => (props) => (...args) => {
  const allClass = args.map((arg) => {
    if (Array.isArray(arg)) {
      const dinamicClasses = getDynamicClasses(cssModule, props, arg)
      // console.log('>>>dinamic', dinamicClasses)
      return dinamicClasses
    } else if (typeof arg === 'string') {
      const moduleClass = getModuleClasses(cssModule, arg)
      // console.log('>>>module', moduleClass)
      return moduleClass
    } else if (typeof arg === 'object') {
      const objectClass = getObjectClasses(cssModule, arg)
      // console.log('>>>object', objectClass)
      return objectClass
    }
  })
  // console.log(allClass)

  const result = classNames(allClass)

  // console.log(result)
  return result
}
