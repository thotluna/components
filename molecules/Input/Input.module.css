.input-container {
  width: 100%;
}

.is-compact {
  max-width: max-content;
}

.input-frame {
  display: flex;
  width: 100%;
  align-items: center;
  justify-content: space-between;
  padding: 12px 17px;
  border: 2px solid var(--color-primary);
  background-color: var(--color-base-white);
  border-radius: var(--border-radius-full);
}

.input-label {
  font-size: var(--font-size-sm);
}

.is-error {
  border: 2px solid var(--color-secondary);
  color: var(--color-font-error);
}

.input-label.is-error {
  color: var(--color-font-error);
}

.input {
  width: 100%;
  flex: 3;
  border: none;
  border-radius: var(--border-radius-full);
  color: var(--color-font-primary);
  font-family: var(--font-family-sans);
  font-size: var(--font-size-sm);
  outline: none;
}

.input-frame:focus-within {
  box-shadow: var(--box-shadow-in), var(--box-shadow-out);
}

.input.is-error {
  color: var(--color-font-error);
}

.input-error {
  display: flex;
  width: 100%;
  justify-content: flex-end;
}

.input::placeholder {
  color: var(--color-font-base);
}

.input-message {
  display: flex;
  width: max-content;
  flex: 1;
  align-items: center;
  justify-content: center;
  padding: 0.2em 0;
  background-color: var(--color-primary);
  border-radius: var(--border-radius-full);
}

.input-wrap {
  height: 74px;
}
