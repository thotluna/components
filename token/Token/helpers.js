import toKebabCase from '../../utils/toKebalCase'

export const getTokenItems = (tokens, tokenKey) => (group) => {
  if (group) {
    return Object.keys(tokens[group]).map((key) => ({
      label: `--${toKebabCase(tokenKey)}-${group}-${toKebabCase(key)}`,
      value: tokens[group][key],
    }))
  }

  return Object.keys(tokens).map((key) => ({
    label: `--${toKebabCase(tokenKey)}-${toKebabCase(key)}`,
    value: tokens[key],
  }))
}
