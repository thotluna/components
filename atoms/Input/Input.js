import React from 'react'
import PropTypes from 'prop-types'
import styles from './Input.module.css'
import { options } from './constants'
import withStyles from '../../hocs/withStyles'

export const Input = ({
  type,
  value,
  isInline,
  onChange,
  refInput,
  placeholder,
  getStyles,
  label,
  isHideLabel,
  id,
}) => {
  return (
    <label className={getStyles('label')}>
      <span
        className={getStyles('span', {
          'is-hide': isHideLabel,
        })}
      >
        {label}
      </span>
      <input
        className={getStyles('input', {
          'is-inline': isInline,
        })}
        id={id}
        aria-label={label}
        type={type}
        value={value}
        placeholder={placeholder}
        onChange={onChange}
        ref={refInput}
      />
    </label>
  )
}

Input.propTypes = {
  onChange: PropTypes.func.isRequired,
  getStyles: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.oneOf(options.types).isRequired,
  id: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  refInput: PropTypes.any,
  isInline: PropTypes.bool,
  isHideLabel: PropTypes.bool,
}

Input.defaultProps = {
  placeholder: '',
  value: '',
  isInline: false,
  isHideLabel: true,
  onChange: () => {},
  getStyles: () => {},
}

export default withStyles(styles)(Input)
