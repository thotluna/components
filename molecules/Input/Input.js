import React, { forwardRef } from 'react'
import PropTypes from 'prop-types'
import styles from './Input.module.css'
import { options } from './constants'
import { getClasses } from '../../helpers/styles'

import Title from '../../atoms/Title'
import Icon from '../../atoms/Icon'

export const Input = forwardRef(
  (
    {
      id,
      label,
      isShowLabel,
      placeholder,
      value,
      handelOnChange,
      color,
      keyEnterMessage,
      handelOnKeyPress,
      errorMessage,
      isError,
      isCompact,
    },
    ref
  ) => {
    const getStyles = getClasses(styles)({ color })

    const handelOnEnter = (e) => {
      if (e.code === 'Enter') {
        handelOnKeyPress()
      }
    }

    return (
      <div
        className={getStyles('input-container', { 'is-compact': isCompact })}
      >
        {isShowLabel && (
          <label
            htmlFor={id}
            aria-labelledby={id}
            className={getStyles('input-label', ['color'], {
              'is-error': isError,
            })}
          >
            {label}
          </label>
        )}
        <div className={getStyles('input-wrap')}>
          <div
            className={getStyles('input-frame', ['color'], {
              'is-error': isError,
            })}
          >
            <input
              id={id}
              name={id}
              aria-label={id}
              className={getStyles('input', ['color'])}
              placeholder={placeholder}
              value={value}
              onChange={handelOnChange}
              onKeyPress={handelOnEnter}
              ref={ref}
            />
            {isError && <Icon type="alert" color="error" size="sm" />}
            {!isError && !isCompact && keyEnterMessage && (
              <div className={getStyles('input-message')}>
                <Title size="sm" color="primary">
                  {keyEnterMessage}
                </Title>
              </div>
            )}
          </div>
          {isError && (
            <div className={getStyles('input-error')}>
              <Title size="sm" color="error">
                {errorMessage}
              </Title>
            </div>
          )}
        </div>
      </div>
    )
  }
)

Input.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  isCompact: PropTypes.bool,
  isShowLabel: PropTypes.bool,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  handelOnChange: PropTypes.func,
  color: PropTypes.oneOf(options.colors),
  keyEnterMessage: PropTypes.string,
  handelOnKeyPress: PropTypes.func,
  errorMessage: PropTypes.string,
  isError: PropTypes.bool,
}

Input.defaultProps = {}

export default Input
