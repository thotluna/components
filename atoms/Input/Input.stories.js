import React from 'react'
import Input, { options } from './index'

import { getVariusTemplate, getOptionsArgTypes } from '../../helpers/storybook'

const Template = (args) => <Input {...args} />
const ListTemplate = getVariusTemplate(Input)

export default {
  title: 'Atoms/Input',
  component: Input,
  args: {
    id: '#1',
    defaultValue: 'testing',
    placeholder: 'testing',
    label: 'Input',
    type: 'text',
  },
  argTypes: {
    type: getOptionsArgTypes(options.types),
  },
}

export const Default = Template.bind({})

export const Types = ListTemplate.bind({})
Types.args = {
  items: options.types.map((type) => ({
    id: `type-${type}`,
    type,
  })),
}

export const Inline = Template.bind({})
Inline.args = {
  id: '#4',
  isInline: true,
}

export const isNotHideLabel = Template.bind({})
isNotHideLabel.args = {
  isHideLabel: false,
  id: '#5',
}
