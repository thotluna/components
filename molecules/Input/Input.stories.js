import React from 'react'
import Input from './Input'

const Template = (args) => <Input {...args} />

export default {
  title: 'Molecules/Input',
  input: Input,
  args: {
    id: 'input',
    label: 'input',
    placeholder: 'is placeholder',
    isShowLabel: true,
    isError: false,
    errorMessage: 'this is a message error',
  },
  argTypes: {
    value: { control: 'text' },
    isError: { control: 'boolean' },
    isShowLabel: { control: 'boolean' },
    isCompact: { control: 'boolean' },
    keyEnterMessage: { control: 'text' },
  },
}

export const Default = Template.bind({})

export const DefaultCompact = Template.bind({})
DefaultCompact.args = {
  isCompact: true,
}

export const DefaultWithoutLabel = Template.bind({})
DefaultWithoutLabel.args = {
  isShowLabel: false,
}

export const DefaultError = Template.bind({})
DefaultError.args = {
  isError: true,
}

export const DefaultErrorWithoutLabel = Template.bind({})
DefaultErrorWithoutLabel.args = {
  isError: true,
  isShowLabel: false,
}

export const DefaultWithKeyEnterMessage = Template.bind({})
DefaultWithKeyEnterMessage.args = {
  keyEnterMessage: 'Enter for continued',
}
