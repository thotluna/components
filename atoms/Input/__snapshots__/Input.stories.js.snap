// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Input Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <label
    className="label"
  >
    <span
      className="span is-hide"
    >
      Input
    </span>
    <input
      aria-label="Input"
      className="input"
      id="#1"
      onChange={[Function]}
      placeholder="testing"
      type="text"
      value="testing"
    />
  </label>
</div>
`;

exports[`Storyshots Atoms/Input Inline 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <label
    className="label"
  >
    <span
      className="span is-hide"
    >
      Input
    </span>
    <input
      aria-label="Input"
      className="input is-inline"
      id="#4"
      onChange={[Function]}
      placeholder="testing"
      type="text"
      value="testing"
    />
  </label>
</div>
`;

exports[`Storyshots Atoms/Input Is Not Hide Label 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <label
    className="label"
  >
    <span
      className="span"
    >
      Input
    </span>
    <input
      aria-label="Input"
      className="input"
      id="#5"
      onChange={[Function]}
      placeholder="testing"
      type="text"
      value="testing"
    />
  </label>
</div>
`;

exports[`Storyshots Atoms/Input Types 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <label
    className="label"
  >
    <span
      className="span is-hide"
    >
      Input
    </span>
    <input
      aria-label="Input"
      className="input"
      id="#3"
      onChange={[Function]}
      placeholder="testing"
      type="text"
      value="testing"
    />
  </label>
</div>
`;

exports[`Storyshots Atoms/Input Value 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <label
    className="label"
  >
    <span
      className="span is-hide"
    >
      Input
    </span>
    <input
      aria-label="Input"
      className="input"
      id="#2"
      onChange={[Function]}
      placeholder="testing"
      type="text"
      value="testing"
    />
  </label>
</div>
`;
